### PR TITLE
fox_versus decompiled, introduce first_diff.py to the project, fixed fox_option NON_MATCHING build

### PR DESCRIPTION
**Pull request by sonicdcer**

### CODE REVIEW

File: src/main/fox_versus.c [hunk 1]
```diff
@@ -0,0 +1,2018 @@
...
+    }
+
+    return 0;
+}
+
+s32 func_800BE564(void) {
+    s32 i;
+    s32 ret = 0;
+
+    if ((sUnlockLandmaster == 0) && (sUnlockOnFoot == 0) || (gVersusStage == VS_STAGE_SECTOR_Z)) {
```
Comment (petrie911) [inline src/main/fox_versus.c:440]:
```suggestion
    if ((!sUnlockLandmaster && !sUnlockOnFoot == 0) || (gVersusStage == VS_STAGE_SECTOR_Z)) {
```
these are booleans

File: src/main/fox_versus.c [hunk 2]
```diff
@@ -0,0 +1,2018 @@
...
+
+    return 0;
+}
+
+bool func_800C0C84(void) {
+    s32 i;
+
+    for (i = 0; i < 4; i++) {
+        if (gControllerPlugged[i] == 1) {
+            if (gControllerPress[i].button & A_BUTTON) {
```
Comment (petrie911) [inline src/main/fox_versus.c:1224]:
```suggestion
        if ((gControllerPlugged[i] == 1) && (gControllerPress[i].button & A_BUTTON)) {
```
these can probably be combined. Also that might want to be `== true`

File: src/main/fox_versus.c [hunk 3]
```diff
@@ -0,0 +1,2018 @@
...
+    }
+
+    // clang-format off
+    for (i = 0; i < 4; i++) { D_801787F8[i] = 150; }
+    // clang-format on
+
+    if (gSaveFile.save.data.planet[SAVE_SLOT_VENOM_2].normalClear == 1) {
+        sUnlockLandmaster = 1;
+    } else {
+        sUnlockLandmaster = 0;
```
Comment (petrie911) [inline src/main/fox_versus.c:1398]:
```suggestion
        sUnlockLandmaster = false;
```
and similarly with the others

File: src/main/fox_versus.c [hunk 4]
```diff
@@ -0,0 +1,2018 @@
...
+            func_800C1368();
+            D_80178754 = 1;
+            func_800C1690();
+            break;
+
+        case 1:
+            if (!func_800C15D8()) {
+                break;
+            }
+            if (func_800C16A0(30)) {
```
Comment (petrie911) [inline src/main/fox_versus.c:1502]:
```suggestion
            if (func_800C15D8() &&  func_800C16A0(30)) {
```
does this work?

File: src/main/fox_versus.c [hunk 5]
```diff
@@ -0,0 +1,2018 @@
...
+            if (gInputPress->button & B_BUTTON) {
+                D_80178754 = 13;
+            } else {
+                func_800C175C();
+            }
+            break;
+    }
+
+    if (((D_80178754 == 8) || (D_80178754 == 9)) || (D_80178754 == 10)) {
+        D_800D4A98 += 1;
```
Comment (petrie911) [inline src/main/fox_versus.c:1731]:
```suggestion
        D_800D4A98++;
```
Does this match?

File: src/main/fox_versus.c [hunk 6]
```diff
@@ -0,0 +1,2018 @@
...
+    if ((fabsf(gBosses[0].obj.pos.x - (this->obj.pos.x + (arg1 * 650.0f))) < 1500.0f) &&
+        (fabsf(gBosses[0].obj.pos.z - (this->obj.pos.z + (arg2 * 650.0f))) < 1500.0f) &&
+        (fabsf(gBosses[0].obj.pos.y - this->obj.pos.y) < 700.0f)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void func_800C2244(Object_2F4* Enm) {
```
Comment (petrie911) [inline src/main/fox_versus.c:1860]: better to use obj2F4 for consistency. It'll make the rename to Actor easier.

File: src/main/fox_versus.c [hunk 7]
```diff
@@ -0,0 +1,2018 @@
...
+                obj->unk_0CE = 24;
+                obj->timer_0C2 = 30;
+
+                Object_SetInfo(&obj->info, obj->obj.id);
+                obj->info.action = (ObjectFunc) func_800C2244;
+                break;
+            }
+        }
+    }
+}
```
Comment (petrie911) [inline src/main/fox_versus.c:2018]:
```suggestion
}

```

File: src/main/fox_versus.c [hunk 8]
```diff
@@ -0,0 +1,2018 @@
...
+    }
+
+    if (D_80178758 == 0) {
+        D_80178758 = 2;
+
+        src.x = 0.0f;
+        src.y = 500.0f;
+        src.z = -10000.0f;
+
+        for (i = 0, obj = &gObjects2F4[10]; i < 5; i++, obj++) {
```
Comment (petrie911) [inline src/main/fox_versus.c:1998]: Same here, `obj2F4` over `obj`

**Review by petrie911 — COMMENTED** [review]
a few things. nothing too major

File: src/main/fox_versus.c (hunk 4 above)
Comment (sonicdcer) [reply src/main/fox_versus.c:1502]: nope :/

**Pull request merged**